mayastor/src/bdev/nexus/nexus_metadata.rs

```diff
@@ -5,821 +5,745 @@
 //!
 //! The data layout is as follows:
 //!  - The first block of the partition is not used.
-//!  - The second block contains a MetaDataHeader (currently 72 bytes) while the
-//!    remainder of the block is padded with zeros.
-//!  - The "index" starts at the third block and contains a fixed number of
-//!    MetaDataIndexEntry entries, each of which contains the address of an
-//!    object that has been written to the partition.
-//!  - The first usable "data" block is the first block following the index
-//!    (whose size is aligned to the blocksize of the disk).
+//!  - The second block contains the MetaDataIndex (currently 80 bytes) while
+//!    the remainder of the block is padded with zeros.
+//!  - The data in the "index" starts at the third block.
+//!  - This and each subsequent block contains a single MetaDataObject entry.
+//!  - The size of a MetaDataObject entry therefore cannot exceed the block_size
+//!    of the device.
+//!  - Once the index is "full", when a new entry is added, the oldest entry is
+//!    (silently) removed in order to make space for the new entry.
```

Comment:
Looks like a ring buffer to me, right? And the metadataindex has a pointer to latest entry into this ring buffer. So when we fill the whole ring, we start overwriting entries from 0 again. That also means that for each change to metadata we have to update two blocks: the index and the entry. Is my understanding right?

Reply:
Correct. Each change will involve writing out a new entry, and also updating the block containing the index.

Reply:
thanks for confirmation 👍

Reply:
what's the idea behind the ring buffer?